### PR TITLE
unattended, consumer, ub24

### PR DESCRIPTION
**Pull request by trel**
this doesn't work yet - live local server is not being detected during setup.

### CODE REVIEW

File: irods_catalog_provider/entrypoint.sh
Comment (korydraughn) [inline irods_catalog_provider/entrypoint.sh:15]: Is this intended or just in a half-way state?

Comment (trel) [reply irods_catalog_provider/entrypoint.sh:15]: it was so i could get in and run the setup step manually... to see the 'attempting' messages.

```
controller.py: 150 - Attempting to connect to iRODS server on port 1247. Attempt #100
```

Comment (korydraughn) [reply irods_catalog_provider/entrypoint.sh:15]: That makes sense.

**Comment by trel** [comment]
ha, silly computer.  i figured it out.

ready for eyeballs.

File: irods_catalog_provider/entrypoint.sh
Comment (alanking) [inline irods_catalog_provider/entrypoint.sh:20]: I think we need to keep something like this around. With these changes, the entrypoint script will attempt to setup the server every time the container starts (I think), which of course will only work the first time. Can you confirm this?

If that's true, we can do basically the same thing, keying off of the `/unattended_install.json` file instead of `/irods_setup.input`.

**Review by alanking — COMMENTED** [review]
Looks fine. I have one concern over restart-ability with these changes, but I think it's easy enough to resolve.

File: irods_catalog_provider/entrypoint.sh
Comment (trel) [reply irods_catalog_provider/entrypoint.sh:20]: Hmm, well, the /unattended_install.json will always be there... in its current state.

We could easily produce a *new* json via the `sed` replacement... and then we can use the 'new' file as the conditional for the initial logic...

Comment (alanking) [reply irods_catalog_provider/entrypoint.sh:20]: The current input file is acting both as the input for the setup script and as an indicator that setup needs to be run. When the image is built, we `COPY` the file in and the entrypoint will run setup the first time it starts because the input file exists, and then it removes the input file afterwards which means setup will not be run again (unless we intentionally create that file in that container again).

We can now do the same thing for the unattended JSON input file. In that way, the JSON file will only be there the first time... We would remove it after setup completes. Similarly, the sed replacement could happen inside the if statement, only the first time. We wouldn't need to any of it after that.

File: irods_catalog_consumer/Dockerfile
Comment (alanking) [inline irods_catalog_consumer/Dockerfile:1]: Consider changing this to at least 22.04. Don't have to.

Comment (trel) [reply irods_catalog_consumer/Dockerfile:1]: Tried it. Required more changes - so will separate that upgrade.

**Comment by trel** [comment]
please to look - all squashed and ready

**Review by korydraughn — COMMENTED** [review]
Seems good.

What's the reason for the switch to unattended install?
Does it work / stand-up?

**Comment by trel** [comment]
Less maintenance over time - and it's the preferred/recommended way to stand up iRODS.  The single-file to stdin to answer the irods_setup.py questions was always a workaround.

Yes, it all stands up and works.  This is mergeable unless you find something glaring.

**Review by korydraughn — APPROVED** [review]
I didn't spot anything.

Let's get one more approval before pounding / merging.

**Review by alanking — APPROVED** [review]
Yep, let's get this in

**Comment by trel** [comment]
all #'d

**Pull request merged**
